test/ViprTests/ViprUnitTests/Given_the_Vipr_Bootstrapper.cs

```diff
@@ -0,0 +1,173 @@
+using Microsoft.Its.Recipes;
+using Microsoft.MockService;
+using Moq;
+using Moq.Protected;
+using Vipr;
+using Vipr.Core;
+using Vipr.Core.CodeModel;
+using Xunit;
+
+namespace ViprUnitTests
+{
+    public class Given_the_Vipr_Bootstrapper
+    {
+        private readonly Mock<Bootstrapper> _bootstrapperMock;
+        private readonly Mock<IOdcmReader> _readerMock;
+        private readonly Mock<IOdcmWriter> _writerMock;
+        private readonly string _workingDirectory;
+
+        public Given_the_Vipr_Bootstrapper()
+        {
+            _bootstrapperMock = new Mock<Bootstrapper>(MockBehavior.Strict) { CallBase = true };
+            _readerMock = new Mock<IOdcmReader>(MockBehavior.Strict);
+            _writerMock = new Mock<IOdcmWriter>(MockBehavior.Strict);
+            _workingDirectory = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+        }
+
+        [Fact]
+        public void When_edmx_is_on_filesystem_it_is_given_to_the_reader_and_the_writers_output_is_written_to_disk()
+        {
+            var inputFiles = Any.FileAndContentsDictionary(1);
+            var metadataPath = Path.Combine(_workingDirectory, inputFiles.First().Key);
+            var metadata = inputFiles.First().Value;
+
+            WriteAsTextFiles(inputFiles);
+
+            var commandLine = String.Format("compile fromedmx {0} to {1}", metadataPath, _workingDirectory);
+
+            ValidateProxyGeneration(metadata, commandLine);
+
+            DeleteFiles(inputFiles);
+        }
+
+        [Fact]
+        public void When_edmx_is_in_cloud_it_is_given_to_the_reader_and_the_writers_output_is_written_to_disk()
+        {
+            var metadataPath = Any.UriPath(1);
+            var metadata = Any.String(1);
+
+            using (var mockService = new MockService()
+                .Setup(r => r.Request.Path.ToString() == "/" + metadataPath &&
+                            r.Request.Method == "GET",
+                       r => r.Response.Write(metadata))
+                .Start())
+            {
+                var metadataUri = mockService.GetBaseAddress() + metadataPath;
+                var commandLine = String.Format("compile fromweb {0} to {1}", metadataUri, _workingDirectory);
+
+                ValidateProxyGeneration(metadata, commandLine);
+            }
+        }
+
+        [Fact]
+        public void When_loading_from_edmx_it_exports_the_model()
+        {
+            var inputFiles = Any.FileAndContentsDictionary(1);
+            var metadataPath = Path.Combine(_workingDirectory, inputFiles.First().Key);
+            var metadata = inputFiles.First().Value;
+            var exportPath = Path.Combine(_workingDirectory, Any.String());
+
+            WriteAsTextFiles(inputFiles);
+
+            var commandLine = String.Format("compile fromedmx {0} to {1} --modelexport {2}", metadataPath, _workingDirectory, exportPath);
+
+            ValidateProxyGeneration(metadata, commandLine);
+
+            DeleteFiles(inputFiles);
```

Comment:
Should this (and the following File.Delete() call) be done in a try/finally block?